test/command/cache_test.rb

```diff
@@ -3,44 +3,67 @@
 
 describe Licensed::Command::Cache do
   let(:config) { Licensed::Configuration.new }
+  let(:source) { TestSource.new }
   let(:generator) { Licensed::Command::Cache.new(config) }
+  let(:fixtures) { File.expand_path("../../fixtures", __FILE__) }
 
   before do
     config.ui.level = "silent"
-    FileUtils.rm_rf config.cache_path
+    config.apps.each do |app|
+      app.sources.clear
+      app.sources << source
+    end
   end
 
-  it "extracts license info for each ruby dep" do
-    generator.run
-    assert config.cache_path.join("rubygem/licensee.txt").exist?
-    license = Licensed::License.read(config.cache_path.join("rubygem/licensee.txt"))
-    assert_equal "licensee", license["name"]
-    assert_equal "mit", license["license"]
+  after do
+    config.apps.each do |app|
+      FileUtils.rm_rf app.cache_path
+    end
+  end
+
+  each_source do |source_type|
```

Comment:
the configuration and expectations for each dependency source is loaded from the new yaml fixtures.  This seemed like the easiest and cleanest way to make these tests dynamic to the requirements for each dependency source